packages/core/src/types.ts

```diff
@@ -58,6 +58,36 @@ export interface LogoutUrlResponse {
   data: { url: LogoutUrl };
 }
 
+/**
+ * A Promise to a response that can be awaited like normal,
+ * but at the risk of not getting all results due to API limits.
+ * In which case a nextCursor field is provided to request the next page.
+ * Helper methods are provided to abstract away the pagination.
+ *
+ * Example using `client.timeseries.list`:
+ * ```js
+ * const response = client.timeseries.list({ filter: { assetIds: [ASSET_ID] }, limit: 1000 });
+ * ```
+ *
+ * You can get up to 1000 elements with normal await:
+ * ```js
+ * const items = await response;
```

Comment:
```suggestion
 * const { items, nextCursor } = await response;
```
I believe it's like this

Reply:
Hmm, can one actually await a CursorAndAsyncIterator? It doesn't have a then function (?)

Reply:
it does `CursorAndAsyncIterator<T> = Promise<ListResponse<T[]>> & CogniteAsyncIterator<T>`